eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -328,7 +328,14 @@ private void publish(BinlogEntry entry, BinlogEntryHandler binlogEntryHandler, B
     this.timeOfFirstMessage.compareAndSet(0, timeNow);
     this.timeOfLatestMessage.set(timeNow);
 
-    CompletableFuture<?> publishingFuture = binlogEntryHandler.publish(entry);
+
+    CompletableFuture<?> publishingFuture = null;
+    try {
+      publishingFuture = binlogEntryHandler.publish(entry);
+    } catch (Exception e) {
```

Comment:
This looks a bit unusual: publish() returns a CompletableFuture.
Shouldn't the exception be within the completableFuture - not thrown outside of the future?
Also, if the CompletableFuture contains an exception, does that get passed to handleProcessingFailException?